lib/components/user/notification-prefs-pane.js

```diff
@@ -28,85 +36,393 @@ const allowedNotificationChannels = [
+                {showPhoneError && <HelpBlock>{INVALID_PHONE_MSG}</HelpBlock>}
+              </ControlStrip>
+            </FormGroup>
+          ) : (
+            <FormGroup>
+              <ControlLabel>SMS notifications will be sent to:</ControlLabel>
+              <ControlStrip>
+                <InlineStatic>
+                  {isPending
+                    // eslint-disable-next-line jsx-a11y/label-has-for
+                    ? <>{formatPhoneNumber(newPhoneNumber)} <Label bsStyle='warning'>Pending</Label></>
+                    // eslint-disable-next-line jsx-a11y/label-has-for
+                    : <>{formatPhoneNumber(currentNumber)} {currentNumberVerified && <Label bsStyle='success'>Verified</Label>}</>
+                  }
+                </InlineStatic>
+                <Button onClick={this._handleEditNumber}>Change number</Button>
+              </ControlStrip>
+            </FormGroup>
+          )}
+
+        {isPending && (
+          // FIXME: If removing the Save/Cancel buttons on the account screen,
+          // make this <FormGroup> a <form> and remove onKeyDown handler.
+          <FormGroup validationState={codeErrorState}>
+            <p>
+              Please check the SMS messaging app on your mobile phone
+              for a text message with a verification code, and enter the code below.
+            </p>
+            <ControlLabel>Verification code:</ControlLabel>
+            <ControlStrip>
+              <Field as={InlineTextInput}
+                name='validationCode'
+                onKeyDown={this._handleValidationCodeKeyDown}
+                placeholder='123456'
+                // onBlur, onChange, and value are passed automatically by Formik
+              />
+              <Button
+                bsStyle='primary'
+                disabled={!!errors.validationCode}
+                onClick={this._handleSubmitCode}
               >
-                {text}
-              </ToggleButton>
-            ))}
-          </ToggleButtonGroup>
-        </ButtonToolbar>
-      </FormGroup>
-      <Details>
-        {notificationChannel === 'email' && (
-          <FormGroup>
-            <ControlLabel>Notification emails will be sent out to:</ControlLabel>
-            <FormControl readOnly type='text' value={email} />
-          </FormGroup>
-        )}
-        {notificationChannel === 'sms' && (
-          // FIXME: Merge the validation feedback upon approving PR #224.
-          <FormGroup validationState={phoneValidationState}>
-            <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
-            <FormControl
-              name='phoneNumber'
-              onBlur={handleBlur}
-              onChange={handleChange}
-              type='tel'
-              value={phoneNumber}
-            />
-            <FormControl.Feedback />
-            {errors.phoneNumber && <HelpBlock>{errors.phoneNumber}</HelpBlock>}
+                Verify
+              </Button>
```

Comment:
With commit 6052d6f04f4fa72753788a2c49f8d9071ad4c36e on otp-middleware, I get this error when I click verify:

![image](https://user-images.githubusercontent.com/2370911/96904759-89bc3f00-1465-11eb-9949-67d212db942c.png)

Reply:
That is from a missing `!` (`!isPhoneNumberValidE164(...)` in the backend...

Reply:
Should be addressed by using https://github.com/ibi-group/otp-middleware/pull/83/commits/6088b315d3e47d419c6f388cca00ad5c51ab942f.